src/debug/GdbStub.h

```diff
@@ -0,0 +1,156 @@
+
+#ifndef GDBSTUB_H_
+#define GDBSTUB_H_
+
+#include <stddef.h>
+#include <map>
+#include <vector>
+
+#include "../types.h"
+
+#include "GdbArch.h"
+
+namespace Gdb {
+
+enum class TgtStatus {
+	NoEvent,
+
+	None,
+	Running,
+	SingleStep,
+	BreakReq, // "break" command from gdb client
+	Bkpt,
+	Watchpt,
+	BkptInsn, // "bkpt" instruction
+	FaultData, // data abort
+	FaultIAcc, // instruction fetch abort
+	FaultInsn, // illegal instruction
+};
+
+struct StubCallbacks {
```

Comment:
can you make this an interface instead (with pure virtual methods instead of function pointers)? Then ARM could just implement that.